app/components/Application/ApplicationDetailsCardItemCustomFields.tsx

```diff
@@ -117,10 +117,24 @@ const CUSTOM_FIELDS: Record<string, React.FunctionComponent<FieldProps>> = {
   emissionSource: EmissionSourceFields,
   emissionGas: (props) => <EmissionGasFields {...props} />,
   product: (props) => (
-    <ProductField query={props.formContext.query} {...props} />
+    <ProductField
+      naicsCode={
+        props.formContext.formResult
+          .applicationRevisionByApplicationIdAndVersionNumber.naicsCode
```

Comment:
I don't see a `formResult` in the `formContext`.
This is actually exactly why I wrote the other comment, about not using `formContext`, and having those fields that need a fragment in created in the `ApplicationDetailsCardItemComponent`

Reply:
I've moved the custom_fields around in `ApplicationDetailsCardItem` & `Form` to remove reliance on the formContext object. I also created a tech_debt card to fix the remaining warning on the allProducts fragment in Form.tsx: https://youtrack.button.is/issue/GGIRCS-2372